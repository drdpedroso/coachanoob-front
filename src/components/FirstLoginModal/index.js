import React from 'react'
import PropTypes from 'prop-types';
import {Header, Image, Modal} from 'semantic-ui-react'
import FirstLoginForm from '../FirstLoginForm'
import logo from '../../images/logo.png'
import styled from "styled-components";

const ImageStyled = styled(Image)`
    margin: auto;
`

const HeaderContainer = styled.div`
    text-align: center;
    margin-top: 20px;
`

const FirstLoginModal = (props) => {
    return (
        <Modal trigger={props.openModal}>
            <Modal.Content>
                <ImageStyled src={logo} size='small'/>
                <HeaderContainer>{props.header && <Header as="h1">{props.header}</Header>}</HeaderContainer>
                <FirstLoginForm/>
            </Modal.Content>
        </Modal>
    )
};

FirstLoginModal.propTypes = {
    openModal: PropTypes.node,
    header: PropTypes.string,
    children: PropTypes.node
};

export default FirstLoginModal
