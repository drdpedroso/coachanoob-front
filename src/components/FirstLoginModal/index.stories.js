import React from 'react';
import {storiesOf} from '@storybook/react';

import FirstLoginModal from './';
import {Button} from "semantic-ui-react"


storiesOf('FirstLoginModal', module)
    .add('with default', () => (
        <FirstLoginModal openModal={<Button>Open Modal</Button>}/>
    ))

    .add('with header', () => (
        <FirstLoginModal header="Bem Vindo, meu parceiro" openModal={<Button>Open Modal</Button>}/>
    ))
