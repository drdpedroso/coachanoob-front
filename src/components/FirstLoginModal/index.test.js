import React from 'react';
import FirstLoginModal from './index'
import renderer from 'react-test-renderer';
import {Button} from "semantic-ui-react";

it('renders correctly', () => {
    const tree = renderer
        .create(<FirstLoginModal openModal={<Button>Open Modal</Button>}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with header', () => {
    const tree = renderer
        .create(<FirstLoginModal header="I'm the header" openModal={<Button>Open Modal</Button>}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
