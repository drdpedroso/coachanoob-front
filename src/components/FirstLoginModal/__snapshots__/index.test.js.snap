// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="ui button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Open Modal
</button>
`;

exports[`renders correctly with header 1`] = `
<button
  className="ui button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Open Modal
</button>
`;
